### PR TITLE
Added solution for Project Euler problem 38.

### PR DESCRIPTION
**Pull request by fpringle**
Name: Pandigital multiples

Take the number 192 and multiply it by each of 1, 2, and 3:

192 × 1 = 192
192 × 2 = 384
192 × 3 = 576
By concatenating each product we get the 1 to 9 pandigital, 192384576. We will call 192384576 the concatenated product of 192 and (1,2,3)

The same can be achieved by starting with 9 and multiplying by 1, 2, 3, 4, and 5, giving the pandigital, 918273645, which is the concatenated product of 9 and (1,2,3,4,5).

What is the largest 1 to 9 pandigital 9-digit number that can be formed as the concatenated product of an integer with (1,2, ... , n) where n > 1?

Reference: [https://projecteuler.net/problem=38](url)
Reference: #2695

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by dhruvmanila — COMMENTED** [review]
### Announcement:

@fpringle

*This message is sent to all pull requests which are **OPEN, NOT MERGED, not containing 'wontfix' label and contains 'Project Euler' or 'Project' or 'Euler' (case doesn't matter) in their title**. If this message doesn't apply to your pull request, please ignore this.*

### Message:

This is to notify all the users submitting a pull request for the Project Euler solution that the [README.md](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) file has been updated with all the style requirements necessary for the directory and solution files. ***Please read through it and make all the necessary changes required to your submission.***

File: project_euler/problem_038/sol1.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+(1,2,3,4,5).
+
+What is the largest 1 to 9 pandigital 9-digit number that can be formed as the
+concatenated product of an integer with (1,2, ... , n) where n > 1?
+"""
+
+
+def is_9_palindromic(n: int) -> bool:
+    """
+    Checks whether n is a 9-digit 1 to 9 pandigital number.
```
Comment (dhruvmanila) [inline project_euler/problem_038/sol1.py:24]: Does it check whether `n` is pandigital or palindromic? Please update either the docstring or function name accordingly.

File: project_euler/problem_038/sol1.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        if is_9_palindromic(candidate):
+            return candidate
+
+    for base_num in range(333, 99, -1):
+        candidate = 1002003 * base_num
+        if is_9_palindromic(candidate):
+            return candidate
+
+    # just in case
+    return 192384576
```
Comment (dhruvmanila) [inline project_euler/problem_038/sol1.py:73]:
```suggestion
```
Please remove this. If your solution gives the correct answer, this is not needed.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: project_euler/problem_038/sol1.py (hunk 2 above)
Comment (fpringle) [reply project_euler/problem_038/sol1.py:73]: If I remove these lines then mypy throws an error "Missing return statement". Is that a problem?

File: project_euler/problem_038/sol1.py (hunk 1 above)
Comment (fpringle) [reply project_euler/problem_038/sol1.py:24]: Sorry, you're right - the function name is wrong.

File: project_euler/problem_038/sol1.py (hunk 2 above)
Comment (dhruvmanila) [reply project_euler/problem_038/sol1.py:73]: Please put `return None`.
Refer: https://github.com/python/mypy/issues/3974#issuecomment-698360152

File: project_euler/problem_038/sol1.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    s = str(n)
+    return len(s) == 9 and set(s) == set("123456789")
+
+
+def solution() -> Union[int, None]:
+    """
+    Return the largest 1 to 9 pandigital 9-digital number that can be formed as the
+    concatenated product of an integer with (1,2,...,n) where n > 1.
+
+    Solution:
```
Comment (dhruvmanila) [inline project_euler/problem_038/sol1.py:45]: Please put the solution explanation in the module-level docstring.

File: project_euler/problem_038/sol1.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+The same can be achieved by starting with 9 and multiplying by 1, 2, 3, 4, and 5,
+giving the pandigital, 918273645, which is the concatenated product of 9 and
+(1,2,3,4,5).
+
+What is the largest 1 to 9 pandigital 9-digit number that can be formed as the
+concatenated product of an integer with (1,2, ... , n) where n > 1?
+"""
+
+from __future__ import annotations
```
Comment (dhruvmanila) [inline project_euler/problem_038/sol1.py:21]: I don't think you're using this anywhere.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

**Review by dhruvmanila — APPROVED** [review]

**Pull request merged**